framework/log/Logger.php

```diff
@@ -311,4 +332,22 @@ public static function getLevelName($level)
 
         return isset($levels[$level]) ? $levels[$level] : 'unknown';
     }
+
+    /**
+     * Calculates the elapsed time for the given profile end token.
+     * @param string $endToken token for the code block.
+     * @param float $time the profile end timestamp.
+     * @return float the elapsed time in seconds.
+     */
+    private function calculateProfileDuration($endToken, $time)
+    {
+        foreach ($this->_profileBeginnings as $i => $beginning) {
```

Comment:
as far as I see you could index the `$this->_profileBeginnings` array by `$token` and then select the necessary token to calculate timing instaead of iterating over the array. Or am I missing something?